### PR TITLE
Release 3.1.2

### PR DESCRIPTION
**Pull request by pauarge**
Releasing 3.1.2, which introduces some minor code improvements in preparation for 3.2.0.

This branch cherry-picks some changes from develop (see commit history) and puts them into trunk. We are including the CHANGELOG and version bumps alongside.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -12,24 +12,31 @@
...
+/CONTRIBUTING.md                 export-ignore
+/composer.json                   export-ignore
+/composer.lock                   export-ignore
+/package.json                    export-ignore
+/package-lock.json               export-ignore
+/phpstan.neon                    export-ignore
+/phpunit.xml.dist                export-ignore
+/phpunit-integration.xml.dist    export-ignore
+/webpack.config.js               export-ignore
+/WINDOWS.md               		 export-ignore
```
Comment (acicovic) [inline .gitattributes]: We seem to have an alignment issue here compared to the above lines?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,16 +1,38 @@
...
 
 Please search the [repo's issues](https://github.com/Parsely/wp-parsely/issues) to see if your issue has been reported already and if so, comment on that issue instead of opening a new one.
 
 When creating a new issue, please add specific steps to reproduce the problem, upload any relevant screenshots, and describe what happened and what you expected would happen instead.
 
-## Setting up your environment
+## Contributing code
 
-### Minimum required versions
+You are welcome to contribute to the plugin by submitting Pull Requests fix issues or introduce new features to the Parse.ly plugin.
```
Comment (acicovic) [inline CONTRIBUTING.md]:
```suggestion
You are welcome to contribute to the plugin by submitting Pull Requests that fix issues or introduce new features to the Parse.ly plugin.
```

File: .gitattributes (hunk 1 above)
Comment (GaryJones) [reply .gitattributes]: Tabs are being used for some reason.
Screenshot 2022-02-24 at 00 30 41

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,30 @@ All notable changes to this project will be documented in this file.
...
+- Compatibility issues when the default category wasn't `Uncategorized`. [#620](https://github.com/Parsely/wp-parsely/pull/620)
+- Fixed some type safety issues. [#657](https://github.com/Parsely/wp-parsely/pull/657)
+- Update outdated references in function comments. [#676](https://github.com/Parsely/wp-parsely/pull/676)
+- Exclude some files & dirs from exported plugin archives. [#612](https://github.com/Parsely/wp-parsely/pull/612)
+- Not installing wp-env globally on CI. [#665](https://github.com/Parsely/wp-parsely/pull/665)
+- README.md: Change "meta data" to "metadata". [#614](https://github.com/Parsely/wp-parsely/pull/614)
+- Update CONTRIBUTING.md for clarity and readability. [#613](https://github.com/Parsely/wp-parsely/pull/613)
+- Adding help for local development issues. [#616](https://github.com/Parsely/wp-parsely/pull/616)
+- Improve documentation. [#654](https://github.com/Parsely/wp-parsely/pull/654)
+- Minor additions to the contributing file. [#683](https://github.com/Parsely/wp-parsely/pull/683)
```
Comment (GaryJones) [inline CHANGELOG.md]: I would consolidate all of these down to a single "Improve documentation" entry; we don't need 5 entries for fixing up docs.

**Review by acicovic — APPROVED** [review]

**Pull request merged**